components/locid/src/parser/langid.rs

```diff
@@ -155,17 +158,17 @@ pub const fn parse_language_identifier_without_variants_from_iter(
             if let Ok(s) = subtags::Region::from_bytes_manual_slice(t, start, end) {
                 region = Some(s);
                 position = ParserPosition::Variant;
-            } else if subtags::Variant::from_bytes_manual_slice(t, start, end).is_ok() {
-                // We cannot handle variants in a const context
-                return Err(ParserError::InvalidSubtag);
+            } else if let Ok(v) = subtags::Variant::from_bytes_manual_slice(t, start, end) {
+                variant = Some(v);
+                break;
```

Comment:
Issue: I think you should not call `break;` because that will allow things like `locale!("und-variant1-variant2")` to get through (the second variant will be skipped). Instead, you should set the state to `ParserPosition::Variant`, and in that state, you can fail if you encounter a second variant (`if variant.is_some()`).

Please also add a test for this behavior. You can use `compile_fail` to test code that isn't supposed to compile:

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html

Reply:
Done